src/calendar/CalendarMonth.jsx

```diff
@@ -30,11 +24,39 @@ const CalendarMonth = React.createClass({
     onMonthChange: React.PropTypes.func,
     onYearChange: React.PropTypes.func,
     value: CustomPropTypes.momentOrMomentRange,
+    locale: React.PropTypes.string,
+  },
+
+  _setupLocale(locale) {
+    this.moment = moment();
+
+    if (locale !== 'en') {
+      require(`moment/locale/${locale}`);
+    }
+
+    this.moment.locale(locale);
+
+    const lang = this.moment.localeData(locale);
+
+    this.WEEKDAYS = Immutable.List(lang._weekdays).zip(Immutable.List(lang._weekdaysShort));
+    this.MONTHS = Immutable.List(lang._months);
+  },
+
+  componentWillMount() {
+    const { locale } = this.props;
+    this._setupLocale(locale);
+  },
+
+  componentWillReceiveProps(nextProps) {
+    const { locale } = nextProps;
+    if (locale !== this.props.locale) {
+      this._setupLocale(locale);
+    }
```

Comment:
If we compute the weekday/month names on the fly, we can simplify this and get rid of `componentWillMount`, `componentWillReceiveProps`, and `_setupLocale`.

For example: https://github.com/onefinestay/react-daterange-picker/pull/56/files#diff-78a3b23346ce557395cf4cc9eca5d165R78